src/SourceConstraints.hs

```diff
@@ -179,6 +180,20 @@ sortedIEs dynFlags ies =
       Module . render dynFlags $ unLoc moduleName
     ieClass ie = error $ "Unsupported: " <> gshow ie
 
+sortedIEThingWith :: DynFlags -> IE GhcPs -> Maybe SDoc
+sortedIEThingWith dynFlags (IEThingWith _xIE _wrappedName _ieWildcard ieWith _ieFieldLabels) =
+  if rendered /= expected
+     then pure $ text . message $ intercalate ", " expected
+     else empty
+  where
+    message :: String -> String
+    message example = "Unsorted import/export with list: (" <> example <> ")"
```

Comment:
@mbj I find the "with list" wording a bit confusing. What about just

* Unsorted import/export list: ...

OT (and non-blocking): Is there a way to actually be precise and refer to import **or** export?

Reply:
> OT (and non-blocking): Is there a way to actually be precise and refer to import **or** export?

Its the same AST node, and I do not have the context on which side we are (import or export)!

Reply:
OT: and source constraints would flag on both (import or export!).

Reply:
>     * Unsorted import/export list: ...

Well its not an import export list, its a `with` list, see the data type we match here. I think we should stick to the GHC AST terminology. Its a clear difference if we have an unsorted list, or an unsorted list inside a namesapce (and we do not yet know if we are dealing with record fields or constructors, this is not known at this stage!).

Reply:
What about "Unsorted import/export detected: ..." then? I still think the "with" in there is too confusing "grammar" for a human to interpret, even if it matches HS internals most closely.